hardware/gpt-utils/gpt-utils.cpp

```diff
@@ -1072,12 +1072,14 @@ static int get_dev_path_from_partition_name(const char *partname,
                 if (stat(path, &st)) {
                         goto error;
                 }
-                if (readlink(path, buf, buflen) < 0)
-                {
+                char resolved[PATH_MAX] = {0};
+                if (!realpath(path, resolved)) {
                         goto error;
-                } else {
-                        buf[PATH_TRUNCATE_LOC] = '\0';
                 }
+                int pos = (int)strlen(resolved) - 1;
+                for (; pos >= 0 && isdigit(resolved[pos]); pos--) ;
```

Comment:
Since this loop has no body and no initializer, I think it's generally more desirable to write `while (pos >= 0 && isdigit(resolved[pos])) pos--;`.

Reply:
preparing now new PR, but I'll reply here.

yes, that would make sense. I'll make this change before submitting